idl/github.com/uber/cadence/matching.thrift

```diff
@@ -33,6 +33,7 @@ struct PollForDecisionTaskResponse {
   30: optional shared.WorkflowType workflowType
   40: optional i64 (js.type = "Long") previousStartedEventId
   50: optional i64 (js.type = "Long") startedEventId
+  60: optional shared.WorkflowQuery query
```

Comment:
I don't like this approach as it feels hacky.  Looks like we need a new API which is more generic and has the capability to return 2 different type of tasks.

Reply:
as discussed offline, we will stick to this approach for now.